packages/csharp/PortingAssistant/PortingAssistant.Common/Utils/TelemetryCollectionUtils.cs

```diff
@@ -80,10 +82,85 @@ public static ProjectMetrics createProjectMetric(string runId, string triggerTyp
                 NumReferences = projectAnalysisResult.ProjectReferences.Count,
                 IsBuildFailed = projectAnalysisResult.IsBuildFailed,
                 CompatibilityResult = projectAnalysisResult.ProjectCompatibilityResult,
-                PortingAssistantVersion = MetricsBase.Version
+                PreCompatibilityResult = GenerateCompatibilityResults(preTriggerData?.sourceFileAnalysisResults?.ToList(),
+                    preTriggerData?.projectPath, preTriggerData?.ported),
+                PortingAssistantVersion = MetricsBase.Version,
+                PreApiInCompatibilityCount = preTriggerData?.incompatibleApis,
+                PostApiInCompatibilityCount = GetIncompatibleApiCount(projectAnalysisResult),
```

Comment:
Same q as above. Why do we have to recalculate this? Isn't this already calculated in the client?